enhancements/machine-config/mco-flattened-ignition.md

```diff
@@ -0,0 +1,176 @@
+---
+title: MCO Support flattened ignition config
+authors:
+  - "@hardys"
+reviewers:
+  - "@celebdor"
+  - "@cgwalters"
+  - "@crawford"
+  - "@kirankt"
+  - "@runcom"
+  - "@stbenjam"
+approvers:
+  - "@cgwalters"
+  - "@crawford"
+  - "@runcom"
+
+creation-date: 2020-08-21
+last-updated: 2020-09-02
+status: **provisional**|implementable|implemented|deferred|rejected|withdrawn|replaced
+---
+
+# MCO Support flattened ignition config
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+This enhancement proposes adding an interface to the MCO such that a flattened
+ignition config, that is a combination of the managed pointer config and the
+MCO rendered config (inlined via a [data URL](https://tools.ietf.org/html/rfc2397))
+is available.
+
+## Motivation
+
+In some situations it is necessary to perform network configuration before it
+is possible to download the rendered config from the MCS in the ramdisk where
+ignition evaluates the pointer config.
+
+This leads to a chicken-egg problem, where a user could configure the network
+via ignition, but ignition cannot apply any config until it has downloaded
+any append/merge directives.
+
+This situation is quite common in baremetal environments, e.g a bonded pair of nics
+(where the native vlan is often used for PXE provisioning),
+with the controlplane network configured via a VLAN on the same physical interface.
+
+For UPI deployments, it may be possible to work around this situation when
```

Comment:
I'd just link to https://github.com/openshift/enhancements/blob/master/enhancements/rhcos/static-networking-enhancements.md

(Which also leads to the point that the CoreOS team considers these problems solved via the Live ISO as noted below)

Reply:
Done